### PR TITLE
fix: bugfixes and debug hooks

### PR DESCRIPTION
**Pull request by sg-s**

### CODE REVIEW

File: src/data_hub/axes_callback.js [hunk 1]
```diff
@@ -5,6 +5,7 @@
 const x_data = df[x_select.value];
 const y_data = df[y_select.value];
 const size_data = df[size_select.value];
+const id_data = df[ID.value];
```
Comment (sg-s) [inline src/data_hub/axes_callback.js:8]: need to also pack ID in here

File: src/data_hub/axes_callback.js [hunk 2]
```diff
@@ -16,7 +17,7 @@ const sizes = size_data.map(value => {
 });
 
 // Update the data source
-scatter_source.data = { 'x': x_data, 'y': y_data, 'size': sizes };
+scatter_source.data = { 'x': x_data, 'y': y_data, 'size': sizes, 'id': id_data };
```
Comment (sg-s) [inline src/data_hub/axes_callback.js:20]: this was a bug...changing sizes broke hover

File: src/data_hub/button_callback.js [hunk 3]
```diff
@@ -11,16 +11,6 @@ console.log("Will write to this column:", labelColumn);
...
-    // Reset selections for all the rows in question
-    const resetChanges = rowIds.map(rowId => ({
-        rowId: rowId,
-        fieldChangeEvents: [{
-            columnId: labelColumn,
-            newValue: { selectedOptions: [] } 
-        }]
-    }));
-
-    deeporigin.dataHub.primaryDatabase.editRows({ changes: resetChanges });
```
Comment (sg-s) [inline src/data_hub/button_callback.js:23]: @kennyjwilli this should fix the 500s

File: src/data_hub/plotting.py [hunk 4]
```diff
@@ -193,7 +195,7 @@ def scatter(
             x_select=x_select,
             y_select=y_select,
             size_select=size_select,
-            df=df.to_dict("list"),
+            df=df.reset_index().to_dict("list"),
```
Comment (sg-s) [inline src/data_hub/plotting.py:198]: also include index in the df

File: src/data_hub/plotting.py [hunk 5]
```diff
@@ -204,8 +206,11 @@ def scatter(
     # this updates the value of the slider to the currently
     # hovered point
 
+    if hover_callback_code is None:
+        hover_callback_code = js_code["hover_callback"]
+
     hover_callback = CustomJS(
-        code=js_code["hover_callback"],
+        code=hover_callback_code,
```
Comment (sg-s) [inline src/data_hub/plotting.py:213]: we can now inject JS code at execution time to figure out why hovering is broken

**Review by kennyjwilli — APPROVED** [review]

**Pull request merged**